src/control/server/storage/scm/firmware.go

```diff
@@ -122,6 +133,59 @@ func (p *Provider) QueryFirmware(req FirmwareQueryRequest) (*FirmwareQueryRespon
 	return resp, nil
 }
 
+func (p *Provider) getRequestedModules(requestedUIDs []string, ignoreMissing bool) (storage.ScmModules, error) {
+	modules, err := p.backend.Discover()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(requestedUIDs) == 0 {
+		return modules, nil
+	}
+
+	uniqueUIDs := common.DedupeStringSlice(requestedUIDs)
```

Comment:
Fair point, it probably indicates that they have made a copy/paste error or something like that when entering the inputs. I was leaning toward being fairly liberal with the errors in these filters, but duplicates are a pretty cut-and-dry error to detect.

Reply:
Yeah, particularly when it comes to firmware stuff I think I'd lean toward insisting on correct input rather than assuming we know what's correct and trying to fix things up.

Reply:
Fixed